bot/modals/class_modal.py

```diff
@@ -0,0 +1,196 @@
+            embed.add_field(name='Channel', value=channel.mention)
+            embed.add_field(name='Instructor', value=similar_class.class_professor)
+            embed.add_field(name='Class Title', value=similar_class.full_title(), inline=False)
+            await inter.response.send_message(embed=embed)
+            return True
+
+        # if a channel exists, but the class is archived - publish unarchive event for our similar class
+        if channel and similar_class.class_archived:
+            await self._bot.messenger.publish(Events.on_class_unarchive, inter, similar_class)
+            return True
+
+        # a similar class was found in the database, but the paired channel doesn't exist...
+        # the method should never reach this point though, since class channels that have no existing
+        # channel are purged from the database upon startup of SockBot (as well as in a listener)
+        return False
+
+    def _autofill(self, data: Union[tuple[str, int], discord.TextChannel]) -> None:
+        """
+        Automatically fills in the defaults of our items with our given data.
+        If `tuple[str, int]` is given as our data, we can fill:
+        - major
+        - course_number
+
+        If `discord.TextChannel` is given as our data, we can fill some or all items, given:
+        - The name of the channel is formatted as `MAJR-XXXX-[Professor]`
+        - The topic of the channel is formatted as `[Course Title] - [Course Description]`
+        """
+        if isinstance(data, tuple):
+            prefix, num = data
+            self.major.default = prefix
+            self.course_number.default = num
+            return
+
+        split_topic = data.topic.split('-') if data.topic else []
+        split_name = data.name.split('-')
+
+        # fill in the data from our split channel name, if possible
+        if len(split_name) >= 3:
+            if valid_course_maj(split_name[0]):
+                self.major.default = split_name[0].upper()
+            if valid_course_num(split_name[1]):
+                self.course_number.default = split_name[1].strip()
+            self.professor.default = split_name[-1].strip().capitalize()[:MAX_INSTR_LEN]
+
+        # fill in the data from our split channel topic, if possible
+        if len(split_topic) == 2:
+            self.course_title.default = split_topic[0].title().strip()[:MAX_TITLE_LEN]
+            description = split_topic[1].strip()[:MAX_DESCR_LEN]
+            self.course_description.default = description if len(description) != 0 else 'None'
+
+
+def valid_course_maj(course_maj: str) -> bool:
+    return MIN_MAJOR_LEN <= len(course_maj) <= MAX_MAJOR_LEN
+
+
+def valid_course_num(course_num: Union[str, int]) -> bool:
+    number: int
+    if isinstance(course_num, str):
+        if not course_num.isdigit():
+            return False
+        number = int(course_num)
+    else:
+        number = course_num
+    return MIN_CLASS_NUM <= number <= MAX_CLASS_NUM
```

Comment:
int int int
```suggestion
    number: int
    if isinstance(course_num, str) and not course_num.isdigit():
        return False
        
    return MIN_CLASS_NUM <= int(course_num) <= MAX_CLASS_NUM
```

Reply:
int this, int that, intellectually you are bad (changed, thanks)